### PR TITLE
fix for NUTCH-2238 contributed by ptorrestr

### PR DESCRIPTION
**Pull request by ptorrestr**
Add a new plugin for Elasticsearch 2.x based on indexer-elastic

### CODE REVIEW

File: src/plugin/indexer-elastic2/howto_upgrade_es.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+1. Upgrade elasticsearch dependency in src/plugin/indexer-elastic/ivy.xml
+
+2. Upgrade the Elasticsearch specific dependencies in src/plugin/indexer-elastic/plugin.xml
```
Comment (lewismc) [inline src/plugin/indexer-elastic2/howto_upgrade_es.txt]: Same with this

File: src/plugin/indexer-elastic2/howto_upgrade_es.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+1. Upgrade elasticsearch dependency in src/plugin/indexer-elastic/ivy.xml
```
Comment (lewismc) [inline src/plugin/indexer-elastic2/howto_upgrade_es.txt]: This should be
```
src/plugin/indexer-elastic2/ivy.xml
```

**Comment by lewismc** [comment]
@ptorrestr this is a nice patch. I am slightly concerned about the overlap between the two elastic plugin. I would be happy for your plugin here to replace the existing plugin. What do you think? Thanks for the contribution.

**Comment by ptorrestr** [comment]
@lewismc I updated the files. 

Unfortunately, ES 1.X and 2.X are not compatible.
I tested using ES 1.7 and I found an error similar as mentioned [here](https://discuss.elastic.co/t/elasticsearch-2-0-java-client/35301). 

I don't know which one is more popular right now. I think will be good keep both for a while until ES 2.X become more popular.

**Comment by valencik** [comment]
@ptorrestr Have you tested this with Nutch 1.x or will it only work in 2.x?

**Comment by ptorrestr** [comment]
@valencik I only tried Nutch 2.x

**Comment by chrismattmann** [comment]
@ptorrestr @lewismc what's the solution here. Should we merge to 2.x?

**Comment by lewismc** [comment]
I think we should retire the ES 1.X plugin (comment it out in build.xml) and then have this as the default plugin. I will do some core review later today.

File: src/plugin/indexer-elastic2/src/java/org/apache/nutch/indexwriter/elastic2/ElasticIndexWriter.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    }
+    request.setSource(source);
+
+    // Add this indexing request to a bulk request
+    bulk.add(request);
+    indexedDocs++;
+    bulkDocs++;
+
+    if (bulkDocs >= maxBulkDocs || bulkLength >= maxBulkLength) {
+      LOG.info("Processing bulk request [docs = " + bulkDocs + ", length = "
```
Comment (lewismc) [inline src/plugin/indexer-elastic2/src/java/org/apache/nutch/indexwriter/elastic2/ElasticIndexWriter.java]: Logging could be improved here to use parameterized messages
http://www.slf4j.org/faq.html#logging_performance

File: src/plugin/indexer-elastic2/src/java/org/apache/nutch/indexwriter/elastic2/ElasticIndexWriter.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+      if (actionGet.hasFailures()) {
+        for (BulkItemResponse item : actionGet) {
+          if (item.isFailed()) {
+            throw new RuntimeException("First failure in bulk: "
+                + item.getFailureMessage());
+          }
+        }
+      }
+      long msWaited = System.currentTimeMillis() - beforeWait;
+      LOG.info("Previous took in ms " + actionGet.getTookInMillis()
```
Comment (lewismc) [inline src/plugin/indexer-elastic2/src/java/org/apache/nutch/indexwriter/elastic2/ElasticIndexWriter.java]: Logging could be improved here to use parameterized messages
http://www.slf4j.org/faq.html#logging_performance

File: src/plugin/indexer-elastic2/src/java/org/apache/nutch/indexwriter/elastic2/ElasticIndexWriter.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+      bulk = client.prepareBulk();
+      bulkDocs = 0;
+      bulkLength = 0;
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    // Flush pending requests
+    LOG.info("Processing remaining requests [docs = " + bulkDocs
```
Comment (lewismc) [inline src/plugin/indexer-elastic2/src/java/org/apache/nutch/indexwriter/elastic2/ElasticIndexWriter.java]: Logging could be improved here to use parameterized messages
http://www.slf4j.org/faq.html#logging_performance

**Comment by lewismc** [comment]
@ptorrestr can you please address the logging issues? Can you also please comment out the existing entries in [build.xml](https://github.com/apache/nutch/blob/2.x/src/plugin/build.xml) so that these plugins do not clash. That would be great.

**Comment by ptorrestr** [comment]
@lewismc done. Should I change the name to "elastic"?

**Comment by lewismc** [comment]
No, i think that elastic 2 is good. If someone wishes to use elastic1 then they still can. Thanks I will merge ASAP.
Thanks

**Pull request merged**